core/src/main/java/org/ehcache/core/StatusTransitioner.java

```diff
@@ -220,7 +220,7 @@ public void succeeded() {
     public StateTransitionException failed(Throwable t) {
       if (st.done()) {
         if (t != null) {
-          throw new AssertionError("Throwable cannot be null if Transition is done.");
+          throw new AssertionError("Throwable cannot be thrown if Transition is done: " + t.getMessage());
```

Comment:
I tried, but we cannot in jdk6 :-(
But we could call AssertionError.setStatckTrace(), but I am not sure it's right ?